cli/src/pcluster/resources/custom_resources/custom_resources_code/delete_image_stack.py

```diff
@@ -43,27 +69,7 @@ def handler(event, context):
 
         logger.info("Image status is %s", image_status)
         if image_status == "AVAILABLE":
-            try:
-                # tag EC2 AMI
-                image_id = message_json["outputResources"]["amis"][0]["image"]
-                aws_partition = message_json["arn"].split(":")[1]
-                parent_image = message_json["imageRecipe"]["parentImage"]
-                image_arn = f"arn:{aws_partition}:ec2:{aws_region}::image/{image_id}"
-                logger.info("Tagging EC2 AMI %s", image_arn)
-                tag_client = boto3.client("resourcegroupstaggingapi", config=boto3_config, region_name=aws_region)
-                tag_client.tag_resources(
-                    ResourceARNList=[
-                        image_arn,
-                    ],
-                    Tags={
-                        "parallelcluster:build_status": "available",
-                        "parallelcluster:parent_image": parent_image,
-                    },
-                )
-            except KeyError as e:
-                logger.error("Failed to parse message with exception: %s", e)
-            except ClientError as e:
-                logging.error("Tagging EC2 AMI %s failed with exception: %s", image_arn, e)
+            _tag_ami(message_json)
```

Comment:
Should we only tag build_status=available to the ami with stack including latest test component(TagComponent), or with stack including PclusterComponent. Because I think we shouldn't tag  build_status=available to output AMI of first stage, or AMI built with DisablePclusterComponent: True in DevSettings.

Reply:
we should always tag with `build_status=available` otherwise the commands logic that rely on this tag (build, describe, list) will not going to work properly

Reply:
why we need to display AMIs with DisablePclusterComponent: True in describe, list commands? To retrieve first stage AMIs, we can use awscli with name and tag filter rather than describe command.